#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_show_whole_version.dpatch by Stefan Fritsch <sf@sfritsch.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: print whole version number, not just 0.44

@DPATCH@
diff -urNad sam2p-0.44-13~/Makefile sam2p-0.44-13/Makefile
--- sam2p-0.44-13~/Makefile	2006-04-05 21:14:12.000000000 +0200
+++ sam2p-0.44-13/Makefile	2006-04-05 21:14:29.000000000 +0200
@@ -62,6 +62,9 @@
 xpmc.h: cols2.pl
 	perl -x cols2.pl >xpmc.h
 
+version.h:
+	echo "#define SAM2P_VERSION \"`dpkg-parsechangelog -ldebian.dist/changelog|egrep '^Version:'|sed -e 's/Version: //'`\"" >version.h
+
 #HQ=perl -e'$$_=join"",<STDIN>; s@([^\w\.\/\-])@sprintf"\\%03o",ord$$1@ge; print "\"$$_\"\n"'
 HQ=perl -x hq.pl
 #%.tth: %.tte
diff -urNad sam2p-0.44-13~/sam2p_main.cpp sam2p-0.44-13/sam2p_main.cpp
--- sam2p-0.44-13~/sam2p_main.cpp	2005-08-26 08:27:25.000000000 +0200
+++ sam2p-0.44-13/sam2p_main.cpp	2006-04-05 21:14:12.000000000 +0200
@@ -2,6 +2,7 @@
  * by pts@fazekas.hu at Fri Mar 15 16:11:03 CET 2002
  */
 
+#include "version.h"
 #include "gensi.hpp"
 #include "gensio.hpp"
 #include "image.hpp"
@@ -868,7 +869,7 @@
   Error::long_argv0=argv0==(char const*)NULLP ? "sam2p" : argv0;
   Error::argv0=Files::only_fext(Error::long_argv0);
   Error::tmpargv0="_sam2p_";
-  Error::banner0="sam2p v0.47";
+  Error::banner0="sam2p v" SAM2P_VERSION;
 }
 
 /* --- */
